cli/lsp/client.rs

```diff
@@ -87,10 +119,11 @@ trait ClientTrait: Send + Sync {
     &self,
     params: lsp_custom::RegistryStateNotificationParams,
   ) -> AsyncReturn<()>;
-  fn configuration(
+  fn specifier_configurations(
     &self,
-    items: Vec<lsp::ConfigurationItem>,
-  ) -> AsyncReturn<Result<Vec<serde_json::Value>, AnyError>>;
+    uris: Vec<lsp::Url>,
+  ) -> AsyncReturn<Result<Vec<Result<SpecifierSettings, AnyError>>, AnyError>>;
+  fn workspace_configuration(&self) -> AsyncReturn<Result<Value, AnyError>>;
```

Comment:
Instead of having a low level `configuration` method on the Client, I've split it up into two separate higher level functions. This simplifies the caller code and the REPL implementation of this trait.